projects/igniteui-angular/src/lib/grids/pivot-grid/pivot-grid.component.ts

```diff
@@ -745,9 +745,15 @@ export class IgxPivotGridComponent extends IgxGridBaseDirective implements OnIni
 
     /**
      * @hidden @internal
-     * @deprecated in version 18.2.0. Use the `autoGenerate` property instead.
+     * @deprecated in version 18.2.0. Column re-creation now relies on `autoGenerate` instead.
      */
-    public override shouldGenerate: boolean;
+    public override get shouldGenerate(): boolean {
+        return this.autoGenerate;
+    }
+
+    public override set shouldGenerate(value: boolean) {
+        this.autoGenerate = value;
+    }
```

Comment:
The Pivot grid doesn't really support auto generate (or more specifically it [always generates columns automatically](https://github.com/IgniteUI/igniteui-angular/blob/ef0f767f1dbae95c8e9437cbc796c397fab28e43/projects/igniteui-angular/src/lib/grids/pivot-grid/pivot-grid.component.ts#L1056-L1057)) and both `autoGenerate` and `shouldGenerate` are hidden - a.k.a not supported. So no need for functionality fallback.

Reply:
And since we've started - the deprecation message is perhaps only useful if someone ever discovered the API in the IDE. If we even keep it, the direction for `autoGenerate` is certainly not needed here